staging/src/k8s.io/api/storagemigration/v1alpha1/types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.29
+
+// StorageVersionMigration represents a migration of stored data to the latest
+// storage version.
+type StorageVersionMigration struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Specification of the migration.
+	Spec StorageVersionMigrationSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	// Status of the migration.
+	// +optional
+	Status StorageVersionMigrationStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// Spec of the storage version migration.
+type StorageVersionMigrationSpec struct {
+	// The resource that is being migrated from. The migrator sends requests to
+	// the endpoint serving the resource.
+	// Immutable.
+	OriginResource metav1.GroupVersionResource `json:"originResource" protobuf:"bytes,1,opt,name=originResource"`
```

Comment:
Why change the name?

At a high level, why not make this seamless?  The change appear capricious.

Reply:
There was feedback about it on the KEP - https://github.com/kubernetes/enhancements/pull/4240#discussion_r1344480734

Reply:
> There was feedback about it on the KEP - [kubernetes/enhancements#4240 (comment)](https://github.com/kubernetes/enhancements/pull/4240#discussion_r1344480734)

Is it just a vanity change?  There's no separate "migrate to" concept.

Reply:
agree on not changing the name and matching the CRD json tag of `resource`

Reply:
> Is it just a vanity change? There's no separate "migrate to" concept.

It change was meant to be for being more descriptive.


I'll revert it back to the original name. - cc @jpbetz